src/Fluid/Flu_FixUp_Restrict.cpp

```diff
@@ -166,9 +166,51 @@ void Flu_FixUp_Restrict( const int FaLv, const int SonFluSg, const int FaFluSg,
 #        endif
 #        endif // #ifdef GAMER_DEBUG
 
+#        if ( MODEL == ELBDM )
+//       average phase instead of real and imaginary part if option OPT__RES_PHASE is on
+         if ( ResFlu && OPT__RES_PHASE && (TVarCC & (_REAL) || TVarCC & (_IMAG)) ) {
+
```

Comment:
I am not sure I understand. In Prepare_PatchData(), there is a sanity check to ensure the phase interpolation can work properly because the necessary fields may not be ready. But in this case, we access the AMR structure directly, so the necessary fields are all there. Do you mean we should throw an error if we only restrict the real or the imaginary part?

Reply:
In that case, I would check
```
if ( ResFlu && OPT__RES_PHASE && (TVarCC & (_REAL) || TVarCC & (_IMAG)) )
   for (int v=0; v<NCOMP_TOTAL; v++)
      if ( TVarCC & (1L<<v) && v > IMAG )    handle error;
```

Reply:
@KunkelAlexander No, that is not what I meant. When `TVarCC` includes `_REAL | _IMAG` but also some extra fields not belonging to `_REAL | _IMAG | _DENS` (e.g., @JeiwHuang recently added a `Phase` field ([here](https://github.com/hyschive/gamer-fork/blob/psidm/src/TestProblem/ELBDM/PlaneWave/Init_TestProb_ELBDM_PlaneWave.cpp#L270-L275)) in the new test problem `PlaneWave`), your implementation will trigger phase restriction and _ignore_ these extra fields. This behavior is unexpected so we need to either fix it or at least, like `Prepare_PatchData()`, raise an error when applying phase restriction with `TVarCC & ~( _REAL | _IMAG | _DENS ) == true`.

Reply:
@KunkelAlexander Thank you! I am afraid I expressed myself awkwardly, but this is what I meant. I will push the updated files in a second. What I do to fix this issue, is the following: `
// update NFluVar and TFluVarIdxList to exclude density, real and imaginary part if phase restriction is on
#  if ( MODEL == ELBDM )
   if ( ResFlu && OPT__RES_PHASE && (TVarCC & (_REAL) || TVarCC & (_IMAG)) ) {
      NFluVar=0;
      for (int v=0; v<NCOMP_TOTAL; v++)
//       only add field if it is neither density nor real or imaginary part
         if ( TVarCC & (1L<<v) && v != DENS && v != REAL && v != IMAG)    TFluVarIdxList[ NFluVar ++ ] = v;
   }
#  endif`

Reply:
Got it! The current version should not have any redundant updates.